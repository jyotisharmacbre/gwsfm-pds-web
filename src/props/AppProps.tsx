export interface IAppProps{
    Theme: any,
    UseStyles: any
}

export interface IHeaderPage
{
    Title:string,
    ActionList?: IBtnActionProps []
}

export interface IBtnActionProps
{
    Title: string,
    Icon: any,
    Css?: any,
    HandleClick:  ()=> void
}