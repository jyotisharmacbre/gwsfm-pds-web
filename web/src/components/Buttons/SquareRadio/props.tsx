export interface ISquareRadioProps{
    value:string;
    label:string;
    id:string;
}