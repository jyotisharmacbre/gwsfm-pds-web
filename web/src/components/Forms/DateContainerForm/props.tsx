export interface IDateContainerProps{
dateOneLabel: string;
dateTwoLabel: string;
}