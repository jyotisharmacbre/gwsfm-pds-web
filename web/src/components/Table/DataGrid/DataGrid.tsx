import React from 'react';
import BootstrapTable from 'react-bootstrap-table-next';
import paginationFactory, { PaginationProvider, SizePerPageDropdownStandalone } from 'react-bootstrap-table2-paginator';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faExclamationTriangle
} from '@fortawesome/free-solid-svg-icons';
import { IGridTableProps } from '../../../models/IGridTableProps';
import { CircularProgress } from '@material-ui/core';
import excelIcon from '../../../assests/images/excel_icon_white.svg';
import excelIcon2 from '../../../assests/images/excel_icon.svg';

const DataGrid: React.FC<IGridTableProps> = props => {

  const getPaginationButton = (p: any, onPageChange: any) => {
    switch (p.page) {
      case "First":
      case "Last":
      case "Back":
      case "Next":
        return <a key={p.page} className={p.disabled ? 'disabled' : ''} onClick={() => onPageChange(p.page)}></a>;
      default:
        return <a key={p.page} className={p.active ? 'active' : ''} onClick={() => {
          if (props.onTableChange)
            props.onTableChange('pagination', {
              page: p.page,
              sizePerPage: props.queryParams?.pagingParams?.pageSize,
              sortField: props.queryParams?.sortingParams?.sortColumnName,
              sortOrder: props.queryParams?.sortingParams?.sortOrder
            });
        }}>
          {p.page}
        </a>;
    }
  }

  const pageListRenderer = ({
    pages,
    onPageChange
  }) => {
    return (
      props.pagination && <div className="pagination_outer">
        <div>
          {
            pages.map(p => {
              return getPaginationButton(p, onPageChange);
            })
          }
        </div>
      </div>
    );
  };

  const sizePerPageRenderer = ({ options, currSizePerPage, onSizePerPageChange }) => (
    <div className="form_style custom-size-filter">
      <div className="row align-items-center mx-1">
        <div className="col-md-4">
          <span> {props.intl.formatMessage({ id: "LABEL_GRID_TOTAL_NO_OF_RECORD" })} : {props.totalSize}</span>
        </div>
        <div className="col-md-8">
          <div className="form-group m-0 float-md-right d-flex align-items-center">
            <span>{props.intl.formatMessage({ id: "LABEL_GRID_RESULT_PER_PAGE" })}:</span>
            <div className="select-wrapper record-select">
              <select className="form-control"
                onChange={(event) => onSizePerPageChange(event.target.value)}
                value={props.queryParams?.pagingParams?.pageSize}>
                {
                  options.map((option) => (
                    <option key={option.text} className="form-control" >
                      {option.text}
                    </option>
                  ))
                }
              </select>
            </div>
            {
              props.showExcel && <div className="filters_outer m-0 justify-content-between d-flex">
                <span className="m-0">
                  <button
                    className="excel_icon"
                    type="button"
                    onClick={() => props.exportToExcelPipelineData && props.exportToExcelPipelineData()}
                    disabled={props.exportLoader}
                    data-test="export_to_excel"
                  >
                    EXPORT
                    {props.exportLoader && <CircularProgress />}
                    <img className="before_hover" src={excelIcon} alt="microsoft excel icon" />
                    <img className="hover" src={excelIcon2} alt="microsoft excel icon" />
                  </button>
                </span>
              </div>
            }
          </div>
        </div>
      </div>

    </div>
  );

  const emptyDataIndication = () => {
    return (<div className="record_outer">
      <span className="recordNot_found">
        <FontAwesomeIcon icon={faExclamationTriangle} />
        {props.intl.formatMessage({ id: "LABEL_GRID_NO_RECORD_FOUND" })}.
      </span>
    </div>);
  }

  const options = {
    sizePerPageRenderer,                                   // For page size dropdown 
    pageListRenderer,                                      // For pagination list

    page: props.queryParams?.pagingParams?.pageIndex,
    sizePerPage: props.queryParams?.pagingParams?.pageSize,
    totalSize: props.totalSize,
    paginationSize: 5,

    alwaysShowAllBtns: true,
    withFirstAndLast: true,
    hidePageListOnlyOnePage: true,

    firstPageText: 'First',
    prePageText: 'Back',
    nextPageText: 'Next',
    lastPageText: 'Last',

    sizePerPageList:
      [
        { text: '20', value: 20 },
        { text: '30', value: 30 },
        { text: '40', value: 40 },
        { text: '50', value: 50 }
      ]
  };

  return (

    <div>
      {
        <PaginationProvider pagination={paginationFactory(options)}>
          {

            ({ paginationProps, paginationTableProps }) => (
              <div>

                {props.pagination && <SizePerPageDropdownStandalone
                  {...paginationProps} />
                }
                <div className="overflowX-show">
                  <BootstrapTable
                    remote
                    keyField="projectId"
                    data={props.data}
                    columns={props.columns}
                    classes={`${props.className} table_responsive`}
                    onTableChange={props.onTableChange}
                    noDataIndication={emptyDataIndication}
                    sort={{
                      dataField: props.queryParams?.sortingParams?.sortColumnName,
                      order: props.queryParams?.sortingParams?.sortOrder
                    }}
                    {...paginationTableProps}
                  />
                </div>
              </div>
            )
          }
        </PaginationProvider>
      }
    </div>
  );
}

export default DataGrid


