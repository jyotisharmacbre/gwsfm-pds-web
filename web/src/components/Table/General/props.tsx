export interface IGeneralTableProps {
  headers: IGeneralTableHeaderProps[];
  content: string;
  editActionClick: () => void;
}

export interface IGeneralTableHeaderProps {
  heading: string;
  subHeading: string;
}
