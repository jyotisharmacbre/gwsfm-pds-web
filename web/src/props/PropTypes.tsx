export namespace AppTypes
{
    export type Color = 'cbregreen' | 'primary' | 'secondary' | 'save' | 'back'  | undefined;
}