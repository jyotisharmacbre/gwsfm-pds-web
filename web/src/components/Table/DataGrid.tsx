import React from 'react';
import BootstrapTable from 'react-bootstrap-table-next';
import paginationFactory, { PaginationProvider, SizePerPageDropdownStandalone } from 'react-bootstrap-table2-paginator';
import { IGridTableProps } from '../../props/AppProps';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faExclamationTriangle
} from '@fortawesome/free-solid-svg-icons';

const DataGrid: React.FC<IGridTableProps> = props => {

  const getPaginationButton = (p: any, onPageChange: any) => {
    switch (p.page) {
      case "First":
      case "Last":
      case "Back":
      case "Next":
        return <a key={p.page} className={p.disabled ? 'disabled' : ''} onClick={() => onPageChange(p.page)}></a>;

      default:
        return <a key={p.page} className={p.active ? 'active' : ''} onClick={() => onPageChange(p.page)}>{p.page}</a>;
    }
  }

  const pageListRenderer = ({
    pages,
    onPageChange
  }) => {
    return (
      <div className="pagination_outer">
        <div>
          {
            pages.map(p => {
              return getPaginationButton(p, onPageChange);
            })
          }
        </div>
      </div>
    );
  };

  const sizePerPageRenderer = ({ options, currSizePerPage, onSizePerPageChange }) => (
    <div className="form_style py-2 custom-size-filter">
      <div className="row align-items-center mx-1">
        <div className="col-sm-6">
          <span> {props.intl.formatMessage({ id: "LABEL_GRID_TOTAL_NO_OF_RECORD" })} : {props.totalSize}</span>
        </div>
        <div className="col-sm-6">
          <div className="form-group m-0 float-sm-right d-flex align-items-center">
            <span>{props.intl.formatMessage({ id: "LABEL_GRID_RESULT_PER_PAGE" })}:</span>
            <div className="select-wrapper record-select">
              <select className="form-control"
                onChange={(event) => onSizePerPageChange(event.target.value)}
                value={props.queryParams?.pagingParams?.pageSize}>
                {
                  options.map((option) => (
                    <option key={option.text} className="form-control" >
                      {option.text}
                    </option>
                  ))
                }
              </select>
            </div>
          </div>
        </div>
      </div>

    </div>
  );

  const emptyDataIndication = () => {
    return <div className="record_outer"><span className="recordNot_found"><FontAwesomeIcon className="" icon={faExclamationTriangle} />{props.intl.formatMessage({ id: "LABEL_GRID_NO_RECORD_FOUND" })}.</span></div>
  }

  const options = {
    sizePerPageRenderer,                                    // For page size dropdown 
    pageListRenderer,                                      // for pagination list

    page: props.queryParams?.pagingParams?.pageIndex,
    sizePerPage: props.queryParams?.pagingParams?.pageSize,
    totalSize: props.totalSize,
    paginationSize: 5,

    alwaysShowAllBtns: true,
    withFirstAndLast: true,
    hidePageListOnlyOnePage: true,

    firstPageText: 'First',
    prePageText: 'Back',
    nextPageText: 'Next',
    lastPageText: 'Last',

    sizePerPageList:
      [
        { text: '20', value: 20 },
        { text: '30', value: 30 },
        { text: '40', value: 40 },
        { text: '50', value: 50 }
      ]
  };

  return (

    <div>
      {
        <PaginationProvider pagination={paginationFactory(options)}>
          {

            ({ paginationProps, paginationTableProps }) => (
              <div>
                <SizePerPageDropdownStandalone
                  {...paginationProps} />
                <div className="overflowX-show">
                  <BootstrapTable
                    remote
                    keyField="projectId"
                    data={props.data}
                    columns={props.columns}
                    defaultSorted={props.defaultSorted}
                    classes={`${props.className} table_responsive`}
                    onTableChange={props.onTableChange}
                    noDataIndication={emptyDataIndication}
                    {...paginationTableProps} />
                </div>
              </div>
            )
          }
        </PaginationProvider>
      }
    </div>
  );
}

export default DataGrid


