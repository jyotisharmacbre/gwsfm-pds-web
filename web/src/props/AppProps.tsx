import { AppTypes } from "./PropTypes";
import { IProjectForm, IProjectFormAddAction } from "../session/ProjectForm/Type";

export interface IAppProps {
    Theme: any,
    UseStyles: any
}

export interface IHeaderPageProps {
    Title: string,
    ActionList?: IBtnActionProps[]
}

export interface IBtnActionProps {
    Title: string,
    Icon?: any,
    Color?: AppTypes.Color,
    HandleClick: () => void
}

export interface IProjectFormState {
    readonly projectname: string,
    readonly company: string;
    readonly customer_contract: string,
    readonly pmexperience: boolean;
    readonly projectmanager: string;
    readonly projectscope: string;
    readonly locale: string;
    invalidProjectName: boolean;
    invalidProjectManager: boolean;
    invalidCompany: boolean;
    invalidCustomerContract: boolean;
    invalidPMExperience: boolean;
    invalidProjectScope: boolean;
    invalidLocale: boolean;
}

export interface IProjectFormProps {
    handleClick: (data: any) => void;
    addToForm: (data: IProjectForm) => Promise<IProjectFormAddAction>;
    form: IProjectForm;
}
