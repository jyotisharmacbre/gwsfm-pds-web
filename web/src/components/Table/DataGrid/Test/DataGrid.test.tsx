import React from 'react';
import { IntlProvider } from 'react-intl';
import translations from '../../../../Translations/translation';
import { mount } from 'enzyme';
import DataGrid from '../DataGrid';
import { IGridTableProps } from '../../../../props/AppProps';
import { gridTestColumns, gridTestData } from './DataGridTestData';

let wrapper: any;

const mountDataGridTable = (props) => {
    wrapper = mount(
        <IntlProvider locale="en" messages={translations['en'].messages}>
            <DataGrid {...props} />
        </IntlProvider>
    );
};


describe('DataGrid Table tests', () => {
    let Props: IGridTableProps = {
        columns: gridTestColumns(),
        data: gridTestData(),
        sorting: true,
        className: "price-table",
        ActionList: [],
        onTableChange: jest.fn(),
        totalSize: 2,
        pagination: true,
        queryParams: {
            pagingParams: {
                pageIndex: 1,
                pageSize: 2
            },
            sortingParams: {
                sortColumnName: "name",
                sortOrder: 'desc'
            }
        },
        intl: {
            formatMessage: ({ defaultMessage }) => defaultMessage
        },

    };

    beforeEach(() => {
        mountDataGridTable(Props);
    });

    it('defines the component', () => {
        expect(wrapper).toBeDefined();
    });

    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should not show error row when no data is present', () => {
        expect(wrapper.find('.recordNot_found').length).toEqual(0);
    });

    it('should have pagination button count 5', () => {
        expect(wrapper.find('.pagination_outer a').length).toEqual(5);
    });

    it('should not show the size per dropdown element and show total records when pagiantion is off', () => {
        let props = { ...Props };
        props.pagination = false
        mountDataGridTable(props);
        expect(wrapper.find('Component .custom-size-filter').length).toEqual(0);
    });

    it('should show error row when no data is present', () => {
        let props = { ...Props };
        props.data = [];
        mountDataGridTable(props);
        expect(wrapper.find('.recordNot_found').length).toEqual(1);
    });

    it('should have total no of filter page size option equals to 4', () => {
        wrapper.find('.record-select').first().find('option').length
    });
});