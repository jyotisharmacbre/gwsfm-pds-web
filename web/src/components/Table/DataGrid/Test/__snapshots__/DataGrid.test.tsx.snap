// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataGrid Table tests should match the snapshot 1`] = `ReactWrapper {}`;
